src/common/ConnectionConfig.tsx

```diff
@@ -0,0 +1,238 @@
+import React, { PureComponent } from 'react';
+import { InlineFormLabel, LegacyForms, Button } from '@grafana/ui';
+const { Select, Input } = LegacyForms;
+import {
+  DataSourcePluginOptionsEditorProps,
+  onUpdateDatasourceJsonDataOptionSelect,
+  onUpdateDatasourceResetOption,
+  onUpdateDatasourceJsonDataOption,
+  onUpdateDatasourceSecureJsonDataOption,
+  SelectableValue,
+} from '@grafana/data';
+
+import { AwsDataSourceJsonData, AwsDataSourceSecureJsonData, awsAuthProviderOptions } from './types';
+import { standardRegions } from './regions';
+
+export interface Props extends DataSourcePluginOptionsEditorProps<AwsDataSourceJsonData, AwsDataSourceSecureJsonData> {
+  loadRegions?: () => Promise<string[]>;
+  defaultEndpoint?: string;
+}
+
+export interface State {
+  regions: Array<SelectableValue<string>>;
+}
+
+export default class ConnectionConfig extends PureComponent<Props, State> {
+  state: State = {
+    regions: standardRegions,
+  };
+
+  // componentDidMount() {
+  //   TODO, validate the settings and load regions:
+  //   See:
+  //   https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/cloudwatch/components/ConfigEditor.tsx#L43
```

Comment:
The problem with that code is that I don't think it works most of the time. Until you save the DS and go back to the config page it won't actually find any datasource to work with. So it will only work if you are updating already existing datasource (and will break again if you try to update the name) so I don't think it makes sense to try to copy it here in the current form.

Reply:
Right -- did you have a frontend only solution for x-ray?

Reply:
Just to put here what we talked about yesterday, to do this properly is a bit more complex and right now the static list here seems good enough. In the query editor we fetch the regions from the API and you can override it there so even if aws adds new region people won't be stuck.